### PR TITLE
DCGAN overriding training step- Keras 3 Migration (Only Tensorflow Backend)

### PR DESCRIPTION
**Pull request by sineeli**
DCGAN overriding training step- Keras 3 Migration (Only Tensorflow Backend)  fix it - [KerasCV-Fixit](https://github.com/keras-team/keras-cv/issues/2211)

### CODE REVIEW

File: examples/generative/dcgan_overriding_train_step.py [hunk 1]
```diff
@@ -154,10 +156,10 @@ def train_step(self, real_images):
         )
 
         # Sample random points in the latent space
-        random_latent_vectors = tf.random.normal(shape=(batch_size, self.latent_dim))
+        random_latent_vectors = keras.random.normal(shape=(batch_size, self.latent_dim))
```
Comment (fchollet) [inline examples/generative/dcgan_overriding_train_step.py:159]: Critically, this needs to be seeded with a `SeedGenerator` instance attached to the model.

**Review by fchollet — COMMENTED** [review]
Thank you for the PR!

File: examples/generative/dcgan_overriding_train_step.py (hunk 1 above)
Comment (sineeli) [reply examples/generative/dcgan_overriding_train_step.py]: Sure, made the required changes

File: examples/generative/dcgan_overriding_train_step.py [hunk 2]
```diff
@@ -2,22 +2,26 @@
...
-from tensorflow.keras import layers
+
+from keras import layers
+from keras import ops
 import matplotlib.pyplot as plt
 import os
 import gdown
 from zipfile import ZipFile
 
+seed_generator = keras.random.SeedGenerator(42)
```
Comment (fchollet) [inline examples/generative/dcgan_overriding_train_step.py:23]: The seed generator must be attached to the layer. So you should do:

```python
class GAN(keras.Model):
    def __init__(self, **kwargs):
        ...
        self.seed_generator = keras.random.SeedGenerator(1337)
```

Comment (sineeli) [reply examples/generative/dcgan_overriding_train_step.py]: Got it

File: examples/generative/dcgan_overriding_train_step.py [hunk 3]
```diff
@@ -185,9 +193,13 @@ class GANMonitor(keras.callbacks.Callback):
     def __init__(self, num_img=3, latent_dim=128):
         self.num_img = num_img
         self.latent_dim = latent_dim
+        self.seed_generator = keras.random.SeedGenerator(42)
```
Comment (sineeli) [inline examples/generative/dcgan_overriding_train_step.py:196]: Hope this is correct way apart from in model layer

**Review by fchollet — COMMENTED** [review]
The code looks good! Please regenerate the other files.

**Comment by sineeli** [comment]
Retrained and generated the required files, Thanks!

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
